lib/ui/window/platform_configuration.cc

```diff
@@ -301,18 +320,31 @@ Dart_Handle PlatformConfigurationNativeApi::SendPlatformMessage(
         tonic::DartPersistentValue(dart_state, callback),
         dart_state->GetTaskRunners().GetUITaskRunner(), name);
   }
-  if (Dart_IsNull(data_handle)) {
-    dart_state->platform_configuration()->client()->HandlePlatformMessage(
-        std::make_unique<PlatformMessage>(name, response));
-  } else {
-    tonic::DartByteData data(data_handle);
-    const uint8_t* buffer = static_cast<const uint8_t*>(data.data());
-    dart_state->platform_configuration()->client()->HandlePlatformMessage(
-        std::make_unique<PlatformMessage>(
-            name, fml::MallocMapping::Copy(buffer, data.length_in_bytes()),
-            response));
+  HandlePlatformMessage(dart_state, name, data_handle, response);
+
+  return Dart_Null();
+}
+
+Dart_Handle PlatformConfigurationNativeApi::SendPortPlatformMessage(
+    const std::string& name,
+    Dart_Handle identifier,
+    Dart_Handle send_port,
+    Dart_Handle data_handle) {
+  // This can be executed on any isolate.
+  UIDartState* dart_state = UIDartState::Current();
+
+  int64_t c_send_port = tonic::DartConverter<int64_t>::FromDart(send_port);
+  if (c_send_port == ILLEGAL_PORT) {
+    return tonic::ToDart("Invalid port specified");
   }
 
+  fml::RefPtr<PlatformMessageResponse> response =
+      fml::MakeRefCounted<PlatformMessageResponseDartPort>(
+          c_send_port, tonic::DartConverter<int64_t>::FromDart(identifier),
+          name);
+
+  HandlePlatformMessage(dart_state, name, data_handle, response);
```

Comment:
If the `UIDartState` does not have a `platform_message_handler_`, then this should return an error string so the caller knows that the message was not sent

Reply:
Yea, I agree.  We usually just silently drop things.  Since this is a programmer and not a user error it doesn't make sense to return an error.  I added FML_DLOG since it is an engine bug if we get to that point.

Reply:
Other error cases (such as an invalid port) will return an error string that will cause an exception to be thrown on the Dart side.

If the app calls `SendPortPlatformMessage` without registering the background isolate, then currently this method will do nothing.  I think it would be helpful to throw so the developer will understand why the message was not sent.

Reply:
Ah yea, good point.  I was forgetting that was a public API.

Reply:
Done, made change and added test.